packages/amplify-e2e-tests/src/__tests__/graphql-v2/invalid-input-arguments.test.ts

```diff
@@ -0,0 +1,151 @@
+
+const projectName = 'invalidarguments';
+describe('Invalid arguments should throw an error', () => {
+    let projRoot: string;
+    
+    beforeAll(async () => {
+        projRoot = await createNewProjectDir(projectName);
+        await initJSProjectWithProfile(projRoot, {
+            name: projectName,
+        });
+        const v2Schema = 'model_with_datatypes.graphql';
+        await addApiWithBlankSchemaAndConflictDetection(projRoot, { transformerVersion: 2 });
+        await updateApiSchema(projRoot, projectName, v2Schema);
+        await amplifyPush(projRoot);
+        await configureAmplifyAPI(projRoot);
+    });
+
+    afterAll(async () => {
+        await deleteProject(projRoot);
+        deleteProjectDir(projRoot);
+    });
+
+    it('create mutation should error on invalid input arguments', async () => {
+        const invalid_create_inputs = [
+            { title: 'task 1', description: 'task 1 description', priority: 1, dueDate: 'invalid date' }, // Invalid date argument
+            { title: 'test', description: 'task 1 description', priority: 2, dueDate: 1000000 }, // Invalid date argument
+            { title: 'task 1', description: 'task 1 description', priority: 5.1, dueDate: '2022-01-01T00:00:00.000Z' }, // Passing float in place of int
+            { title: 'task 1', description: 'task 1 description', priority: "invalid", dueDate: '2022-01-01T00:00:00.000Z' }, // Passing string in place of int
+        ];
+        
+        await Promise.all(
+            invalid_create_inputs.map(async (input) => {
+                try {
+                    const createResult = await createTask(input);
+                    expect(false).toBeTruthy();
+                } catch (error) {
+                    expect(error.errors).toBeDefined();
+                    expect(error.errors.length).toBeGreaterThanOrEqual(1);
+                }
+            }),
+        );
+    });
+
+    it('list query should error on invalid filter arguments', async () => {
+        const invalid_filter_inputs = [
+            { priority: { eq: 1.5 } }, // Invalid datatype - passing float for int
+            { priority: { eq: "test" } }, // Invalid datatype - passing string for int
+            { priority: { gt: 1.5 } }, // Invalid datatype - passing float for int
+            { priority: { lt: "test" } }, // Invalid datatype - passing string for int
+            { and: { priority: { lt: "test" } } }, // Invalid datatype - passing string for int with 'and' condition
+            { priority: { lt: "test" } }, // Invalid datatype - passing string for int
+            { priority: { between: [1] } }, // Incorrect number of arguments for between operator
+            { priority: { between: [1, 2.5] } }, // Incorrect argument value for between operator
+            { or: [ { priority: { between: [1, 2.5] } }, { title: { gt: "test" } } ] }, // Incorrect argument value for between operator with 'or' condition
+        ];
+
+        await Promise.all(
```

Comment:
Can you pull this block out to a utlity? Then these tests will just be 'data setup'.test basically.

Reply:
Makes sense. Moved the logic to a separate method.